src/topology/compact_open.lean

```diff
@@ -55,6 +71,20 @@ private lemma is_open_gen {s : set α} (hs : is_compact s) {u : set β} (hu : is
   is_open (compact_open.gen s u) :=
 topological_space.generate_open.basic _ (by dsimp [mem_set_of_eq]; tauto)
 
+/-- The compact-open topology is equal to the infimum, as `s` varies over the compact subsets of
+`α`, of the topologies of uniform convergence on `s`. -/
+lemma compact_open_eq_Inf_uniform_on :
```

Comment:
I note you raised the question of whether this should in fact be the definition. I don't think it matters.

Presumably we'll prove lemmas named `compact_open.is_open_iff` etc. once we try to prove a bit more and hopefully this is how people will use this topology rather than from definitions.

For what it's worth, I think I'd be inclined to leave things as they are but I also think it's entirely reasonable to switch to your definition.

Reply:
I think that makes sense, let's keep the old definition.